bin/Trainer.py

```diff
@@ -215,13 +215,15 @@ def _hearRef(self, body):
     assert body[0] == 'referee', 'Expected referee message.'
     _,ts,event = body
     self._frame = int(ts)
-    if event == 'GOAL':
+    if 'GOAL' in event:
       self._numGoals += 1
       self._numGoalFrames += self._frame - self._lastTrialStart
+      event = 'GOAL'
```

Comment:
I don't like renaming events. Instead lets introduce another variable: 
endOfTrial = False
if 'GOAL' in event:
   ...
   endOfTrial = True
elif ...

Then change line 231 to: 
if endOfTrial:
   ...